src/extension.ts

```diff
@@ -63,19 +66,28 @@ export function activate(context: ExtensionContext) {
     showStatus();
   });
 
-  findDiagnostics(workspace.rootPath);
+  // Run the main routine at activation time as well
+  if (workspaceFolders) {
+    for (const folder of workspaceFolders) {
+      findDiagnostics(folder);
+    }
+  }
+
+  // Finds VSCode diagnostics to display based on a coverage file specified by the search pattern in each workspace folder
+  function findDiagnostics(workspaceFolder: WorkspaceFolder) {
+    let searchPattern = new RelativePattern(workspaceFolder, searchCriteria);
```

Comment:
`const` have a tiny performance benefit over `let`

```suggestion
    const searchPattern = new RelativePattern(workspaceFolder, searchCriteria);
```